addons/controllers/clusterbootstrap_controller_test.go

```diff
@@ -38,298 +46,393 @@ var _ = Describe("ClusterBootstrap Reconciler", func() {
-
-			var gvr schema.GroupVersionResource
-			var object *unstructured.Unstructured
-			var fooPackage *runtanzuv1alpha3.ClusterBootstrapPackage
-			var foobar1Package *runtanzuv1alpha3.ClusterBootstrapPackage
-
-			// Verify providerRef exists and also the cloned provider object with ownerReferences to cluster and ClusterBootstrap
-			Eventually(func() bool {
-				Expect(len(clusterBootstrap.Spec.AdditionalPackages) > 1).To(BeTrue())
-
-				fooPackage = clusterBootstrap.Spec.AdditionalPackages[1]
-				Expect(fooPackage.RefName == "foobar.example.com.1.17.2").To(BeTrue())
-				Expect(*fooPackage.ValuesFrom.ProviderRef.APIGroup == "run.tanzu.vmware.com").To(BeTrue())
-				Expect(fooPackage.ValuesFrom.ProviderRef.Kind == "FooBar").To(BeTrue())
-
-				providerName := fmt.Sprintf("%s-foobar-package", clusterName)
-				Expect(fooPackage.ValuesFrom.ProviderRef.Name == providerName).To(BeTrue())
-
-				gvr = schema.GroupVersionResource{Group: "run.tanzu.vmware.com", Version: "v1alpha1", Resource: "foobars"}
-				var err error
-				object, err = dynamicClient.Resource(gvr).Namespace("default").Get(ctx, providerName, metav1.GetOptions{})
-
-				Expect(err).ToNot(HaveOccurred())
+				}, waitTimeout, pollingInterval).Should(BeTrue())
 
-				var foundClusterOwnerRef bool
-				var foundClusterBootstrapOwnerRef bool
-				var foundLabels bool
-				for _, ownerRef := range object.GetOwnerReferences() {
-					if ownerRef.UID == cluster.UID {
-						foundClusterOwnerRef = true
-					}
-					if ownerRef.UID == clusterBootstrap.UID {
-						foundClusterBootstrapOwnerRef = true
-					}
-				}
-				providerLabels := object.GetLabels()
-				if providerLabels[addontypes.ClusterNameLabel] == clusterName &&
-					providerLabels[addontypes.PackageNameLabel] == fooPackage.RefName {
-					foundLabels = true
-				}
-
-				if foundClusterOwnerRef && foundClusterBootstrapOwnerRef && foundLabels {
-					return true
-				}
-				return false
-			}, waitTimeout, pollingInterval).Should(BeTrue())
-
-			// Verify that secret object mapping to foobar1 package exists with ownerReferences to cluster and ClusterBootstrap
-			Eventually(func() bool {
-				foobar1Package = clusterBootstrap.Spec.AdditionalPackages[0]
-
-				Expect(foobar1Package.RefName == "foobar1.example.com.1.17.2").To(BeTrue())
-				secretName := fmt.Sprintf("%s-foobar1-package", clusterName)
-				Expect(foobar1Package.ValuesFrom.SecretRef == secretName).To(BeTrue())
-
-				s := &v1.Secret{}
-				Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: "default", Name: secretName}, s)).To(Succeed())
-
-				var foundClusterOwnerRef bool
-				var foundClusterBootstrapOwnerRef bool
-				var foundLabels bool
-				var foundType bool
-				for _, ownerRef := range s.GetOwnerReferences() {
-					if ownerRef.UID == cluster.UID {
-						foundClusterOwnerRef = true
+				// TODO
+				By("CNI thing")
+				//// Verify CNI is populated in the cloned object with the value from the cluster variables definitions
```

Comment:
These commented tests will be fixed in a future PR?

Reply:
Those are the tests I am still fixing.

Reply:
done now.